### PR TITLE
Companion for #11981

### PR DESCRIPTION
**Pull request by Szegoo**
https://github.com/paritytech/substrate/pull/11981

cumulus companion: https://github.com/paritytech/cumulus/pull/1563

### CODE REVIEW

**Review by shawntabrizi — APPROVED** [review]

File: runtime/polkadot/src/lib.rs [hunk 1]
```diff
@@ -1144,75 +1144,82 @@ impl Default for ProxyType {
...
-impl InstanceFilter<Call> for ProxyType {
-	fn filter(&self, c: &Call) -> bool {
+impl InstanceFilter<RuntimeCall> for ProxyType {
+	fn filter(&self, c: &RuntimeCall) -> bool {
 		match self {
 			ProxyType::Any => true,
 			ProxyType::NonTransfer => matches!(
 				c,
-				Call::System(..) |
-				Call::Scheduler(..) |
```
Comment (bkchr) [inline runtime/polkadot/src/lib.rs:1154]: This disappeared?

File: runtime/polkadot/src/lib.rs [hunk 2]
```diff
@@ -1144,75 +1144,82 @@ impl Default for ProxyType {
...
-				Call::Babe(..) |
-				Call::Timestamp(..) |
-				Call::Indices(pallet_indices::Call::claim{..}) |
-				Call::Indices(pallet_indices::Call::free{..}) |
-				Call::Indices(pallet_indices::Call::freeze{..}) |
+				RuntimeCall::System(..) |
+				RuntimeCall::Vesting(..) |
```
Comment (bkchr) [inline runtime/polkadot/src/lib.rs:1154]: This did changed as well?

**Review by bkchr — CHANGES_REQUESTED** [review]
I found two things where the proxy definition changed. Please do not reorder or change anything there! (This will make the review also much easier).

And especially don't change anything in the logic.

**Comment by Szegoo** [comment]
> I found two things where the proxy definition changed. Please do not reorder or change anything there! (This will make the review also much easier).
> 
> And especially don't change anything in the logic.

Sorry for that, I didn't notice it, this was not an intended change. Will fix this.

File: runtime/kusama/src/lib.rs [hunk 3]
```diff
@@ -1003,86 +1002,95 @@ impl Default for ProxyType {
 	}
 }
 
-impl InstanceFilter<Call> for ProxyType {
-	fn filter(&self, c: &Call) -> bool {
+impl InstanceFilter<RuntimeCall> for ProxyType {
```
Comment (kianenigma) [inline runtime/kusama/src/lib.rs:1005]: lines to double check by other reviewers.

File: runtime/polkadot/src/lib.rs [hunk 4]
```diff
@@ -136,55 +136,55 @@ pub fn native_version() -> NativeVersion {
 }
 
 pub struct BaseFilter;
-impl Contains<Call> for BaseFilter {
-	fn contains(call: &Call) -> bool {
+impl Contains<RuntimeCall> for BaseFilter {
```
Comment (kianenigma) [inline runtime/polkadot/src/lib.rs:139]: I think this is actually not needed anymore.

**Review by kianenigma — APPROVED** [review]

File: runtime/polkadot/src/lib.rs (hunk 4 above)
Comment (kianenigma) [reply runtime/polkadot/src/lib.rs:139]: https://github.com/paritytech/polkadot/pull/5969

Comment (Szegoo) [reply runtime/polkadot/src/lib.rs:139]: so should I just revert this?

**Review by bkchr — APPROVED** [review]

File: runtime/kusama/src/lib.rs (hunk 3 above)
Comment (shawntabrizi) [reply runtime/kusama/src/lib.rs:990]: what needs to be double checked?

**Comment by shawntabrizi** [comment]
bot merge

**Comment by shawntabrizi** [comment]
bot merge

**Pull request merged**